src/transaction.rs

```diff
@@ -35,6 +35,27 @@ pub struct Transaction<'a> {
     listeners: TxListeners<'a>,
 
     done: bool,
+    on_drop: OnTransactionDrop,
+}
+
+/// An enum representing the possible behavior which a [`Transaction`] may exhibit
+/// when it is dropped.
+///
+/// Note that unlike JavaScript's [`IDBTransaction`][1], this crate's [`Transaction`]
+/// defaults to aborting - i.e., [`OnTransactionDrop::Abort`] - instead of
+/// committing - i.e., [`OnTransactionDrop::Commit`] - the transaction!
+///
+/// [1]: https://developer.mozilla.org/en-US/docs/Web/API/IDBTransaction
+#[derive(Debug, Copy, Clone)]
+pub enum OnTransactionDrop {
+    /// Abort the [`Transaction`] when it is dropped. This is the default
+    /// behavior of [`Transaction`].
+    Abort,
```

Comment:
If this is the default behaviour, I think it's valuable to have this:

```suggestion
#[derive(Clone, Copy, Debug, Default)]
pub enum OnTransactionDrop {
    /// Abort the [`Transaction`] when it is dropped. This is the default
    /// behavior of [`Transaction`].
    #[default]
    Abort,
```

Reply:
commit: [`77cd2ab361227c02b2949faaeb27ee9278b37e3b`](https://github.com/matrix-org/rust-indexed-db/pull/1/commits/77cd2ab361227c02b2949faaeb27ee9278b37e3b)